gapis/api/vulkan/command_buffer_rebuilder.go

```diff
@@ -266,17 +274,93 @@ func rebuildVkCmdBeginRenderPass2KHR(
 	return cleanup, cmd, nil
 }
 
-func rebuildVkCmdEndRenderPass(
+func rebuildVkCmdEndRenderPassCommon(
 	ctx context.Context,
 	cb CommandBuilder,
 	commandBuffer VkCommandBuffer,
 	r *api.GlobalState,
 	s *api.GlobalState,
-	d VkCmdEndRenderPassArgsʳ) (func(), api.Cmd, error) {
+	d VkCmdEndRenderPassCommonArgsʳ) (func(), api.Cmd, error) {
+	switch d.Version() {
+	case RenderPassVersion_Renderpass:
+		return rebuildVkCmdEndRenderPass(ctx, cb, commandBuffer, r, s, d)
+	case RenderPassVersion_Renderpass2:
+		return rebuildVkCmdEndRenderPass2(ctx, cb, commandBuffer, r, s, d)
+	case RenderPassVersion_Renderpass2KHR:
+		return rebuildVkCmdEndRenderPass2KHR(ctx, cb, commandBuffer, r, s, d)
+	default:
+		panic("There should be a renderpass version")
+	}
+}
 
+func rebuildVkCmdEndRenderPass(
+	ctx context.Context,
+	cb CommandBuilder,
+	commandBuffer VkCommandBuffer,
+	r *api.GlobalState,
+	s *api.GlobalState,
+	d VkCmdEndRenderPassCommonArgsʳ) (func(), api.Cmd, error) {
 	return func() {}, cb.VkCmdEndRenderPass(commandBuffer), nil
 }
 
+func rebuildVkCmdEndRenderPass2(
+	ctx context.Context,
+	cb CommandBuilder,
+	commandBuffer VkCommandBuffer,
+	r *api.GlobalState,
+	s *api.GlobalState,
+	d VkCmdEndRenderPassCommonArgsʳ) (func(), api.Cmd, error) {
+	mem := []api.AllocResult{}
+
+	subpassEndInfo := createVkSubpassEndInfo(d.PSubpassEndInfo())
+	subpassEndData := s.AllocDataOrPanic(ctx, subpassEndInfo)
+	mem = append(mem, subpassEndData)
+
+	cleanup := func() {
+		for _, d := range mem {
+			d.Free()
+		}
+	}
+
+	cmd := cb.VkCmdEndRenderPass2(
+		commandBuffer,
+		subpassEndData.Ptr(),
+	)
+	for _, d := range mem {
+		cmd.AddRead(d.Data())
+	}
+	return cleanup, cmd, nil
+}
+
+func rebuildVkCmdEndRenderPass2KHR(
+	ctx context.Context,
+	cb CommandBuilder,
+	commandBuffer VkCommandBuffer,
+	r *api.GlobalState,
+	s *api.GlobalState,
+	d VkCmdEndRenderPassCommonArgsʳ) (func(), api.Cmd, error) {
```

Comment:
Might be nice to typedef func() to CleanupFunction or something, so it's more clear what the first return argument is giving you. Or rather, it's clearer what the purpose of the signatureless function is

Reply:
I've seen this in a few places so extend the comment to all of them too

Reply:
Well, that would require to change the entire file(because there is a pattern). We can still do that but not as a part of this PR.